---
title: "Benchmarking of tools for the analysis of methylation and expression data"
author: "BM"
date: "`r Sys.Date()`"
output:
    prettydoc::html_pretty:
    toc: true
    toc_depth: 4
    number_sections: true  
    theme: tactile
---

# Introduction

# Methodology

Tools were obtained from the tool catalogue **bio.tools**, using the keywords *methylation analysis* and *R package*, were 39 tools were identified. Using the keywords *methylation expression analysis* and *BioConductor*, 18 tools were identified. The lists of tools are as follows. Of these, common tools identified were: BPRtools, MEAL, COHCAP, MassArray, FEM, and NanoMethVIz.

# Tools identified
The following information was collated based on the description of the tool in **bio.tools** 18 tools listed from the specific keywords.

## ELMER

This tool is designed to use DNA methylation and gene expression from a large number of samples to infer a regulatory element landscape and transcription factor network in primary tissues. It builds regulatory networks comparing control and tumor samples (by paring methylation and expression data) and correlating that to nearby probes and transcription factors through genome mapping with genome ranges.

Reference: https://academic.oup.com/bioinformatics/article/35/11/1974/5145133
http://bioconductor.org/packages/release/bioc/html/ELMER.html


## MethylMix

Algorithm implemented to identify hyper/hypomethylated genes for a disease. Based on a beta mixture model to identify methylation states and compares them with the normal DNA methylation state. It uses the Differential Methylation value or DM-value defined as the difference of a methylation state with the normal methylation state. Matched gene expression data is used to identify, besides differential, functional methylation states by focusing on methylation changes that effect gene expression.
The package requires 3 files to be provided: Methylation normal, Methylation cancer and GE cancer. With the 3rd dataset, it looks exclusively at negatively correlated data to identify genes regulated by methylation.
http://bioconductor.org/packages/release/bioc/html/MethylMix.html

References: 
Gevaert 0. MethylMix: an R package for identifying DNA methylation-driven genes. Bioinformatics (Oxford, England). 2015;31(11):1839-41. doi:10.1093/bioinformatics/btv020. 
Gevaert O, Tibshirani R, Plevritis SK. Pancancer analysis of DNA methylation-driven genes using MethylMix. Genome Biology. 2015;16(1):17. doi:10.1186/s13059-014-0579-8.


## GGtools

Software and data for analyses in genetics of gene expression and/or DNA methylation. The package identifies eQTLs from associations of DNA polymorphisms and mRNA abundance by exploring public and private archives of genotyping and expression arrays.

Reference:
Carey VJ, Davis AR, Lawrence MF, Gentleman R, Raby BA (2009). “Data structures and algorithms for analysis of genetics of gene expression with Bioconductor: GGtools 3.x.” Bioinformatics, 25(11), 1447–8. doi: 10.1093/bioinformatics/btp169, http://bioinformatics.oxfordjournals.org/content/25/11/1447.long.
http://bioconductor.org/packages/release/bioc/html/GGtools.html

## COHCAP

This package provides a pipeline to analyze single-nucleotide resolution methylation data (Illumina 450k/EPIC methylation array, targeted BS-Seq, etc.). It provides QC metrics, differential methylation for CpG Sites, differential methylation for CpG Islands, integration with gene expression data, and visualization of methylation values.


http://bioconductor.org/packages/release/bioc/html/COHCAP.html

## MEAL

Package to integrate methylation and expression data. It can also perform methylation or expression analysis alone. Several plotting functionalities are included as well as a new region analysis based on redundancy analysis. Effect of SNPs on a region can also be estimated.
Expression and methylation analyses are correlation based. Additionally, a visualization option is available.

Reference: 
Ruiz-Arenas C, Gonzalez J (2021). MEAL: Perform methylation analysis. R package version 1.20.4.
http://bioconductor.org/packages/release/bioc/html/MEAL.html

## FEM

This package performs a systems-level integrative analysis of DNA methylation and gene expression data. It seeks modules of functionally related genes which exhibit differential promoter DNA methylation and differential expression, where an inverse association between promoter DNA methylation and gene expression is assumed.
Removed from BioConductor in October 2020.

http://http://bioconductor.riken.jp/packages/3.7/bioc/html/FEM.html

## MassArray

This package is designed for the import, quality control, analysis, and visualization of methylation data generated using Sequenom's platform. The tools herein contain a highly detailed amplicon prediction for optimal assay design. Also included are quality control measures of data, such as primer dimer and bisulfite conversion efficiency estimation. Methylation data are calculated using the same algorithms contained in the EpiTyper software package.
http://bioconductor.org/packages/release/bioc/html/MassArray.html

## BPRMeth

This package uses the Binomial Probit Regression likelihood to model methylation profiles and extract higher order features. These features quantitate precisely notions of shape of a methylation profile. Using these higher order features across promoter-proximal regions, we construct a powerful predictor of gene expression. Also, these features are used to cluster proximal-promoter regions using the EM algorithm.
From methylation array data and annotation data, it can provide various results, for example gene expression prediction or clustering of methylation profiles, through modeling algorithms.

Reference: Kapourani CA, Sanguinetti G (2018). “BPRMeth: a flexible Bioconductor package for modelling methylation profiles.” Bioinformatics, 34, 2485-2486. doi: 10.1093/bioinformatics/bty129.
http://bioconductor.org/packages/release/bioc/html/BPRMeth.html

## TCGAbiolinksGUI

A Graphical User Interface to analyze cancer molecular and clinical data. For integrative analysis it works with various data inputs like DEA (Differentially gene Expression Analysis), DMR (Differentially Methylated Regions)  or outputs generated with the ELMER package.

http://bioconductor.org/packages/release/bioc/html/TCGAbiolinksGUI.html

## DSS

R library that performs differential analysis for count-based sequencing data. It detects differentially expressed genes (DEGs) from RNA-seq and differentially methylated loci or regions (DML/DMRs) from bisulfite sequencing (BS-seq). Its core is a new dispersion shrinkage method for estimating the dispersion parameter from Gamma-Poisson or Beta-Binomial distributions.
http://bioconductor.org/packages/release/bioc/html/DSS.html

## SMITE

This package builds on the Epimods framework which facilitates finding weighted subnetworks ("modules") on Illumina Infinium 27k arrays using the SpinGlass algorithm, as implemented in the iGraph package. We have created a class of gene centric annotations associated with p-values and effect sizes and scores from any researchers prior statistical results to find functional modules.
http://bioconductor.org/packages/release/bioc/html/SMITE.html

## canceR

This package is a user friendly interface based on the cgdsr and other modeling packages to explore, compare, and analyse all available cancer data (Clinical data, Gene Mutation, Gene Methylation, Gene Expression, Protein Phosphorylation, Copy Number Alteration) hosted by the Computational Biology Center at Memorial-Sloan-Kettering Cancer Center (MSKCC).
http://bioconductor.org/packages/release/bioc/html/canceR.html

## lumi

This package provides an integrated solution for the Illumina microarray data analysis. It includes functions of Illumina BeadStudio (GenomeStudio) data input, quality control, BeadArray-specific variance stabilization, normalization and gene annotation at the probe level. It also includes the functions of processing Illumina methylation microarrays, especially Illumina Infinium methylation microarrays.
http://bioconductor.org/packages/release/bioc/html/lumi.html

## IWTomics

Implementation of the Interval-Wise Testing (IWT) for omics data. This inferential procedure tests for differences in "Omics" data between two groups of genomic regions (or between a group of genomic regions and a reference center of symmetry), and does not require fixing location and scale at the outset.
http://bioconductor.org/packages/release/bioc/html/IWTomics.html

## M3D

This package identifies statistically significantly differentially methylated regions of CpGs. It uses kernel methods (the Maximum Mean Discrepancy) to measure differences in methylation profiles, and relates these to inter-replicate changes, whilst accounting for variation in coverage profiles.
https://bioconductor.riken.jp/packages/3.1/bioc/html/M3D.html


## Repitools

Tools for the analysis of enrichment-based epigenomic data. Features include summarization and visualization of epigenomic data across promoters according to gene expression context, finding regions of differential methylation/binding, BayMeth for quantifying methylation etc.
http://bioconductor.org/packages/release/bioc/html/Repitools.html

## omicRexposome

It systematizes the association evaluation between exposures and omic data, taking advantage of MultiDataSet for coordinated data management, rexposome for exposome data definition and limma for association testing. Also to perform data integration mixing exposome and omic data using multi co-inherent analysis (omicade4) and multi-canonical correlation analysis (PMA).
http://bioconductor.org/packages/release/bioc/html/omicRexposome.html

## NanoMethViz

An R/Bioconductor package for visualizing long-read methylation data. Visualise methlation data from Oxford Nanopore sequencing. NanoMethViz is a toolkit for visualising methylation data from Oxford Nanopore sequencing. It can be used to explore methylation patterns from reads derived from Oxford Nanopore direct DNA sequencing with methylation called by callers including nanopolish, f5c and megalodon. The plots in this package allow the visualisation of methylation profiles aggregated over experimental groups and across classes of genomic features.
https://bioconductor.org/packages/NanoMethViz


From the 39 tools identified with more generalized keywords:

## EpiMethEx

R package for large-scale integrated analysis in methylation hotspots linked to genetic regulation.
https://github.com/giupardeb/EpiMethEx

## MultiDataSet

Implementation of the BRGE's (Bioinformatic Research Group in Epidemiology from Center for Research in Environmental Epidemiology) dataset and MethylationSet. It is designed for integrating multi omics data sets and MethylationSet to contain normalized methylation data. These package contains base classes for MEAL and rexposome packages.
http://bioconductor.org/packages/release/bioc/html/MultiDataSet.html

## IMIX

A multivariate mixture model approach to integrative analysis of multiple types of omics data. R package for integrative genomics analysis using a multivariate mixture model framework. Ziqiao Wang and Peng Wei. IMIX: A multivariate mixture model approach to integrative analysis of multiple types of omics data. bioRxiv 2020.06.23.167312; doi: https://doi.org/10.1101/2020.06.23.167312.
https://github.com/ziqiaow/IMIX



# Tools selected

Needs testing with our data

```{r}
library(BiocManager)
```

## ELMER
## BPRMeth
## EpiMethEx

## Test results


# Key findings



# References

